ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -180,6 +205,23 @@ private HttpClientRequest configureBasicAuth(final HttpClientRequest request) {
     }
   }
 
+  private static <T> void handleSuccessfulQueryResponse(
+      final HttpClientResponse response,
+      final CompletableFuture<T> cf,
+      final ResponseHandlerSupplier<T> responseHandlerSupplier) {
+    final RecordParser recordParser = RecordParser.newDelimited("\n", response);
+    final QueryResponseHandler<T> responseHandler =
+        responseHandlerSupplier.get(Vertx.currentContext(), recordParser, cf);
+
+    recordParser.handler(responseHandler::handleBodyBuffer);
+    recordParser.endHandler(responseHandler::handleBodyEnd);
+    recordParser.exceptionHandler(responseHandler::handleException);
+  }
+
+  private static void handleSuccessfulCloseQueryResponse(final CompletableFuture<Void> cf) {
```

Comment:
These methods and the one above seem like overkill - I'd probably just put them in the lambda.

Reply:
As above, I've refactored this. Hopefully your concerns have been addressed indirectly.